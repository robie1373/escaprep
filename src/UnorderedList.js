import React, { Component } from 'react';
import ListItem from './ListItem';

class UnorderedList extends Component {
  constructor(props) {
    super(props);
    this.listItems = this.listItems.bind(this);
  }

  listItems = () => {
    var l = [];
    var arrayLength = this.props.items.data.length;
    for (var i = 0; i < arrayLength; i++) {
      //console.log(this.props.items.data[i]);
      for (var key in this.props.items.data[i]) {
        //console.log(key);
        if (typeof this.props.items.data[i][key] !== 'function') {
          const li_key = i.toString() + key;
          l.push((<ListItem key={li_key} column_name={key} value={this.props.items.data[i][key]} />));
        }
      }
    }
    return l;
  }

  render () {
    return ( <ul> {this.listItems()} </ul>);
  }
}

export default UnorderedList;
