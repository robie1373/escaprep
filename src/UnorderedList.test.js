import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UnorderedList from './UnorderedList';

test('snapshot', () => {
  const data = {data: [{"one": 1, "two": 2, "three": 3}]};
  const component = renderer.create(<UnorderedList items={data}/>);

  let tree = component.toJSON()
  ;
  expect(tree).toMatchSnapshot();
});
