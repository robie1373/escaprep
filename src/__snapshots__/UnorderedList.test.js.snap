// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<ul>
   
  <li>
    one
     : 
    1
  </li>
  <li>
    two
     : 
    2
  </li>
  <li>
    three
     : 
    3
  </li>
   
</ul>
`;
